lib/aur-fetch

```diff
@@ -8,88 +8,77 @@ readonly startdir=$PWD
 readonly PS4='+(${BASH_SOURCE}:${LINENO}): ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
 
 # default options
-mode=git recurse=0
+log=directory mode=git recurse=0
 
 git_stat_patch() {
     git --no-pager log -O "$1" --patch --stat '..@{upstream}'
 }
 
-merge_upstream() {
-    # set destination of patch file
-    local pkg=$1 log_dir=$2 output
-
-    if [[ -d $log_dir ]]; then
-        output=$log_dir/$pkg.patch
-    else
-        output=/dev/stdout
-    fi
-
-    # default to showing PKGBUILD diffs first (#399)
-    local orderfile=${GIT_DIR:-$PWD/.git}/orderfile
+upstream_sync() {
+    local pkg=$1 patch=${2:-/dev/stdout}
+    local orderfile=${GIT_DIR:-$PWD/$pkg/.git}/orderfile
 
+    # default to showing PKGBUILD in diffs first (#399)
     if [[ ! -s $orderfile ]]; then
         echo PKGBUILD >"$orderfile"
     fi
 
+    # get last upstream revision
     git fetch -v >&2
 
-    # only print log on upstream changes
     if [[ $(git rev-parse HEAD) != $(git rev-parse '@{upstream}') ]]; then
-        git_stat_patch "$orderfile" >> "$output"
+        # only print log on upstream changes
+        git_stat_patch "$orderfile" >> "$patch"
 
         # discard any local changes (#349)
         git reset --hard "HEAD@{upstream}" >&2
     fi
 }
 
 fetch_git() {
-    local log_dir=$1 pkg uri
+    local uri pkg output_dir=$1
 
     while IFS= read -r uri; do
         pkg=${uri##*/}   # strip path
         pkg=${pkg%%.git} # strip .git suffix
 
         if [[ -d $pkg/.git ]]; then
             # avoid issues with filesystem boundaries (#274)
-            GIT_DIR="$PWD/$pkg"/.git GIT_WORK_TREE="$PWD/$pkg" merge_upstream "$pkg" "$log_dir"
+            GIT_DIR="$PWD/$pkg"/.git GIT_WORK_TREE="$PWD/$pkg" upstream_sync "$pkg" "${output_dir:+$output_dir/$pkg.patch}"
```

Comment:
these `$PWD` variables can be removed. The `git` bug that required it is already fixed.
We may take this chance to get rid of it.